cirq-core/cirq/transformers/transformer_api.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2022 The Cirq Developers
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Defines the API for circuit transformers in Cirq."""
+
+import textwrap
+import functools
+from typing import (
+    Callable,
+    Tuple,
+    Hashable,
+    List,
+    cast,
+    Type,
+    overload,
+    Union,
+    TYPE_CHECKING,
+)
+
+import dataclasses
+import enum
+
+if TYPE_CHECKING:
+    import cirq
+
+
+class LogLevel(enum.Enum):
+    """Different logging resolution options for `cirq.TransformerStatsLogger`.
```

Comment:
This should still be clarified.  Does specifying DEBUG also include INFO and WARNING messages?

Reply:
This is specified below in the docstring of `TransformerLogger.show`, which is used to display the logs.

Reply:
Added another description here as well.